edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/handlers/ConnectionHandler.cs

```diff
@@ -68,15 +68,17 @@ public async Task<Option<IDeviceProxy>> GetDeviceProxyAsync(IIdentity identity)
         {
             using (await this.guard.LockAsync())
```

Comment:
it would not be in general, I chose to lock the dictionary for two reason:
- there is a part where multiple operation is done (ReconcileConnectionAsync), and that works only if nothing else can manipulate the dictionary (e.g. if something removes/add an item during reconcile, then the end result will be wrong). Because that locks the dictionary, for me it seemed safer to lock every usage, even if some cases it would not cause problems.
- the dictionary is a ConcurrentDictionary only because it had nicer interface to use. Because of the lockings explained above, a normal dictionary would be sufficient (but with a more cumbersome interface). So the case is rather than I use a ConcurrentDictionary where it is not necessary than using lock where it is not necessary

Reply:
using ConcurrentDictionary when not actually needed comes with a performance cost, I would not use if not needed.